Update the pagy dependency of madmin

0xRichardH opened the pull request:
madmin and pagy 3+ are not working with ruby 3+, so I submit a pull request to update the pagy dependency of madmin. The reasons for updating are as below:

- As [pagy](https://github.com/ddnexus/pagy#new-in-48) mentioned that the pagy 4+ is for ruby 2.5+ and the pagy 3+ is for ruby <2.5. So we should use pagy 4+ because madmin required ruby version  2.5+.
- The pagy 3+ is not working with ruby 3+, we need upgrade pagy to 4+ in order to use ruby 3+.

<img width="886" alt="image" src="https://user-images.githubusercontent.com/7600503/127148946-6f471847-1b16-4117-b6e3-98d6ae58d68b.png">

<img width="812" alt="image" src="https://user-images.githubusercontent.com/7600503/127148709-0c1aa98c-fef7-4505-ba9d-4cf9a0a873f6.png">

0xRichardH committed:
Upgrade pagy dependency to >= 4.0

https://github.com/ddnexus/pagy#new-in-48

0xRichardH committed:
Merge pull request #1 from Digital-Star-Tech/ruby-3_0_2

Upgrade pagy dependency to >= 4.0

excid3 commented:
Since we're not using hardly any special features of Pagy, I think we could just relax our dependency to support both.

```ruby
spec.add_dependency "pagy", ">= 3.5", "< 5.0"
```

That way we can support the same Ruby versions that Rails does and it will install whichever version of Pagy is needed.

0xRichardH committed:
Relax pagy version dependency.

https://github.com/excid3/madmin/pull/81#issuecomment-887573359

0xRichardH commented:
@excid3  👍 cool, I've made the changes.

excid3 commented:
I think you'll need to run `bundle && bundle exec appraisal` and commit the changes to the `Gemfile.lock` files for the tests to pass.

0xRichardH committed:
Update dependences

```bash
bundle && bundle exec appraisal
```

https://github.com/excid3/madmin/pull/81#issuecomment-887963688

0xRichardH commented:
@excid3 All of the tests have been passed.

<img width="866" alt="image" src="https://user-images.githubusercontent.com/7600503/127356714-a4ab2c42-19f4-4a92-b675-ee69124ff489.png">

excid3 commented:
Thank you!
